models/_sector/depin/geodnet/geodnet_revenue.sql

```diff
@@ -0,0 +1,127 @@
+{{ config(
+    schema = 'geodnet'
+    , alias = 'revenue'
+    , materialized = 'incremental'
+    , file_format = 'delta'
+    , incremental_strategy = 'merge'
+    , unique_key = ['date', 'chain', 'name']
+    )
+}}
+
+WITH
+  hour_series AS (
+    SELECT
+      time AS hour,
+      DATE_TRUNC('day', time) AS day
+    FROM
+      UNNEST (
+        sequence(
+          CAST('2023-04-20 00:00' AS timestamp),
+          CAST(
+            DATE_TRUNC('hour', CURRENT_TIMESTAMP) AS timestamp
+          ),
+          INTERVAL '1' hour
+        )
+      ) AS t (time)
+      {% if is_incremental() %}
+      WHERE
+          time >= (
+            SELECT
+              CAST(COALESCE(MAX(date), '2023-04-27 00:00') AS timestamp) - INTERVAL '7' DAY
+            FROM
+              {{ this }}
+          )
+      {% endif %}
+  ),
+  burn_events AS (
+    SELECT
+      DATE_TRUNC('hour', a.evt_block_time) AS evt_hour,
+      DATE_TRUNC('day', a.evt_block_time) AS evt_day,
+      CAST(value AS double) / pow(10, b.decimals) AS tokens_burned
+    FROM {{ source('erc20_polygon', 'evt_transfer') }} a
+      JOIN {{ source('tokens', 'erc20') }} b ON a.contract_address = b.contract_address
```

Comment:
may want to add `blockchain` to join conditions in case address is repeated across chains